packages/react-components/src/components/EndCallButton.tsx

```diff
@@ -56,13 +74,16 @@ export const EndCallButton = (props: EndCallButtonProps): JSX.Element => {
     return <CallEndIcon key={'callEndIconKey'} />;
   };
 
-  const defaultRenderText = (props?: IButtonProps): JSX.Element => {
-    return (
-      <Label key={'callEndLabelKey'} className={mergeStyles(controlButtonLabelStyles, props?.styles?.label)}>
-        Leave
-      </Label>
-    );
-  };
+  const defaultRenderText = useCallback(
+    (props?: IButtonProps): JSX.Element => {
+      return (
+        <Label key={'callEndLabelKey'} className={mergeStyles(controlButtonLabelStyles, props?.styles?.label)}>
+          {text}
+        </Label>
+      );
+    },
+    [text]
```

Comment:
Should `props?.styles?.label` also be in this dep array?

Reply:
Nope since props is a argument of the callback.
![image](https://user-images.githubusercontent.com/79475487/123307246-1f1d4080-d4d7-11eb-8d79-529298a9f874.png)

Reply:
oh nice!